### ISSUE TITLE
[1.18.2] Server crashes when chased by skeleton

### ISSUE THREAD

**Issue by truemogician** [Yesssssman/epicfightmod#959]
This problem occurs from time to time when a player is chased by a skeleton. Here's the complete crash report:   
```
Description: Ticking entity

java.lang.NullPointerException: Cannot invoke "net.minecraft.world.entity.Entity.m_20185_()" because "p_24961_" is null
	at net.minecraft.world.entity.ai.control.LookControl.m_24960_(LookControl.java:33) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading}
	at yesman.epicfight.world.entity.ai.goal.ChasingGoal.m_8037_(ChasingGoal.java:140) ~[EpicFight-18.0.3-beta.jar%2355!/:1.0] {re:classloading}
	at net.minecraft.world.entity.ai.goal.WrappedGoal.m_8037_(WrappedGoal.java:65) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.entity.ai.goal.GoalSelector.m_186081_(GoalSelector.java:119) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorGoalSelector,pl:mixin:A}
	at net.minecraft.world.entity.ai.goal.GoalSelector.m_25373_(GoalSelector.java:110) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorGoalSelector,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_6140_(Mob.java:698) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorMob,pl:mixin:APP:domesticationinnovation.mixins.json:MobMixin,pl:mixin:A}
	at net.minecraft.world.entity.LivingEntity.m_8107_(LivingEntity.java:2506) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:inventorio.mixins.json:optional.totem.LivingEntityMixin,pl:mixin:APP:botania_xplat.mixins.json:AccessorLivingEntity,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:curios.mixins.json:MixinLivingEntity,pl:mixin:APP:domesticationinnovation.mixins.json:LivingEntityMixin,pl:mixin:APP:create.mixins.json:CustomItemUseEffectsMixin,pl:mixin:APP:create.mixins.json:accessor.LivingEntityAccessor,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_8107_(Mob.java:495) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorMob,pl:mixin:APP:domesticationinnovation.mixins.json:MobMixin,pl:mixin:A}
	at net.minecraft.world.entity.monster.Monster.m_8107_(Monster.java:41) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,re:mixin}
	at net.minecraft.world.entity.monster.Zombie.m_8107_(Zombie.java:235) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading}
	at net.minecraft.world.entity.LivingEntity.m_8119_(LivingEntity.java:2253) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:inventorio.mixins.json:optional.totem.LivingEntityMixin,pl:mixin:APP:botania_xplat.mixins.json:AccessorLivingEntity,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:curios.mixins.json:MixinLivingEntity,pl:mixin:APP:domesticationinnovation.mixins.json:LivingEntityMixin,pl:mixin:APP:create.mixins.json:CustomItemUseEffectsMixin,pl:mixin:APP:create.mixins.json:accessor.LivingEntityAccessor,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_8119_(Mob.java:307) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorMob,pl:mixin:APP:domesticationinnovation.mixins.json:MobMixin,pl:mixin:A}
	at net.minecraft.world.entity.monster.Zombie.m_8119_(Zombie.java:209) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8647_(ServerLevel.java:652) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:MixinServerLevel,pl:mixin:A}
	at net.minecraft.world.level.Level.m_46653_(Level.java:486) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:319) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:MixinServerLevel,pl:mixin:A}
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:299) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:MixinServerLevel,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:902) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:326) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:842) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:685) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) [?:?] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at net.minecraft.world.entity.ai.control.LookControl.m_24960_(LookControl.java:33) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading}
	at yesman.epicfight.world.entity.ai.goal.ChasingGoal.m_8037_(ChasingGoal.java:140) ~[EpicFight-18.0.3-beta.jar%2355!/:1.0] {re:classloading}
	at net.minecraft.world.entity.ai.goal.WrappedGoal.m_8037_(WrappedGoal.java:65) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.entity.ai.goal.GoalSelector.m_186081_(GoalSelector.java:119) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorGoalSelector,pl:mixin:A}
	at net.minecraft.world.entity.ai.goal.GoalSelector.m_25373_(GoalSelector.java:110) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorGoalSelector,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_6140_(Mob.java:698) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorMob,pl:mixin:APP:domesticationinnovation.mixins.json:MobMixin,pl:mixin:A}
	at net.minecraft.world.entity.LivingEntity.m_8107_(LivingEntity.java:2506) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:inventorio.mixins.json:optional.totem.LivingEntityMixin,pl:mixin:APP:botania_xplat.mixins.json:AccessorLivingEntity,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:curios.mixins.json:MixinLivingEntity,pl:mixin:APP:domesticationinnovation.mixins.json:LivingEntityMixin,pl:mixin:APP:create.mixins.json:CustomItemUseEffectsMixin,pl:mixin:APP:create.mixins.json:accessor.LivingEntityAccessor,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_8107_(Mob.java:495) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorMob,pl:mixin:APP:domesticationinnovation.mixins.json:MobMixin,pl:mixin:A}
	at net.minecraft.world.entity.monster.Monster.m_8107_(Monster.java:41) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,re:mixin}
	at net.minecraft.world.entity.monster.Zombie.m_8107_(Zombie.java:235) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading}
	at net.minecraft.world.entity.LivingEntity.m_8119_(LivingEntity.java:2253) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:inventorio.mixins.json:optional.totem.LivingEntityMixin,pl:mixin:APP:botania_xplat.mixins.json:AccessorLivingEntity,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:curios.mixins.json:MixinLivingEntity,pl:mixin:APP:domesticationinnovation.mixins.json:LivingEntityMixin,pl:mixin:APP:create.mixins.json:CustomItemUseEffectsMixin,pl:mixin:APP:create.mixins.json:accessor.LivingEntityAccessor,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_8119_(Mob.java:307) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:AccessorMob,pl:mixin:APP:domesticationinnovation.mixins.json:MobMixin,pl:mixin:A}
	at net.minecraft.world.entity.monster.Zombie.m_8119_(Zombie.java:209) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8647_(ServerLevel.java:652) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:MixinServerLevel,pl:mixin:A}
	at net.minecraft.world.level.Level.m_46653_(Level.java:486) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:319) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:MixinServerLevel,pl:mixin:A}
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:299) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:MixinServerLevel,pl:mixin:A}
-- Entity being ticked --
Details:
	Entity Type: minecraft:zombie (net.minecraft.world.entity.monster.Zombie)
	Entity ID: 13129
	Entity Name: Zombie
	Entity's Exact location: 347.41, 64.00, -445.64
	Entity's Block location: World: (347,64,-446), Section: (at 11,0,2 in 21,4,-28; chunk contains blocks 336,-64,-448 to 351,319,-433), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,-64,-512 to 511,319,-1)
	Entity's Momentum: 0.00, -0.08, 0.00
	Entity's Passengers: []
	Entity's Vehicle: null
Stacktrace:
	at net.minecraft.world.level.Level.m_46653_(Level.java:486) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:319) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:MixinServerLevel,pl:mixin:A}
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:299) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:botania_xplat.mixins.json:MixinServerLevel,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:902) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:326) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:842) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:685) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) [?:?] {}


-- Affected level --
Details:
	All players: 1 total; [ServerPlayer['true_mogician'/519, l='ServerLevel[world]', x=334.06, y=65.92, z=-476.36]]
	Chunk stats: 3013
	Level dimension: minecraft:overworld
	Level spawn location: World: (282,69,-437), Section: (at 10,5,11 in 17,4,-28; chunk contains blocks 272,-64,-448 to 287,319,-433), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,-64,-512 to 511,319,-1)
	Level time: 14466720 game time, 13221425 day time
	Level name: world
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Level weather: Rain time: 1 (now: false), thunder time: 1 (now: false)
	Known server brands: forge
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:902) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:326) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:842) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:685) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[server-1.18.2-20220404.173914-srg.jar%2381!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) [?:?] {}


-- System Details --
Details:
	Minecraft Version: 1.18.2
	Minecraft Version ID: 1.18.2
	Operating System: Windows Server 2022 (amd64) version 10.0
	Java Version: 17.0.2, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode, sharing), Oracle Corporation
	Memory: 316468184 bytes (301 MiB) / 2491416576 bytes (2376 MiB) up to 10737418240 bytes (10240 MiB)
	CPUs: 32
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz
	Identifier: Intel64 Family 6 Model 62 Stepping 4
	Microarchitecture: Ivy Bridge (Server)
	Frequency (GHz): 2.60
	Number of physical packages: 2
	Number of physical CPUs: 16
	Number of logical CPUs: 32
	Graphics card #0 name: Microsoft 基本显示适配器
	Graphics card #0 vendor: (标准显示卡类型) (0x102b)
	Graphics card #0 VRAM (MB): 0.00
	Graphics card #0 deviceId: 0x0534
	Graphics card #0 versionInfo: DriverVersion=10.0.20348.1
	Graphics card #1 name: Microsoft Remote Display Adapter
	Graphics card #1 vendor: Microsoft
	Graphics card #1 VRAM (MB): 0.00
	Graphics card #1 deviceId: unknown
	Graphics card #1 versionInfo: DriverVersion=10.0.20348.617
	Memory slot #0 capacity (MB): 32768.00
	Memory slot #0 clockSpeed (GHz): 1.87
	Memory slot #0 type: DDR3
	Memory slot #1 capacity (MB): 32768.00
	Memory slot #1 clockSpeed (GHz): 1.87
	Memory slot #1 type: DDR3
	Virtual memory max (MB): 75218.54
	Virtual memory used (MB): 15580.16
	Swap memory total (MB): 9728.00
	Swap memory used (MB): 4013.36
	JVM Flags: 1 total; -Xmx10G
	Server Running: true
	Player Count: 1 / 8; [ServerPlayer['true_mogician'/519, l='ServerLevel[world]', x=334.06, y=65.92, z=-476.36]]
	Data Packs: vanilla, mod:farmersdelight (incompatible), mod:grapplemod (incompatible), mod:voicechat (incompatible), mod:sedna (incompatible), mod:aoa3, mod:publicguiannouncement (incompatible), mod:jei (incompatible), mod:ae2 (incompatible), mod:botania (incompatible), mod:wthit (incompatible), mod:flywheel (incompatible), mod:curios (incompatible), mod:bowinfinityfix (incompatible), mod:patchouli (incompatible), mod:create, mod:xaerominimap, mod:waystones (incompatible), mod:gravestone (incompatible), mod:epicfight, mod:xaeroworldmap, mod:markdown_manual (incompatible), mod:citadel (incompatible), mod:autoreglib (incompatible), mod:travelersbackpack (incompatible), mod:storagedrawers (incompatible), mod:oc2 (incompatible), mod:drawerstooltip (incompatible), mod:elevatorid, mod:guardvillagers (incompatible), mod:balm (incompatible), mod:jeresources (incompatible), mod:bettershields (incompatible), mod:cloth_config (incompatible), mod:forge (incompatible), mod:goprone, mod:quark (incompatible), mod:veinmining (incompatible), mod:domesticationinnovation (incompatible), mod:enhancedarmaments (incompatible), mod:cataclysm (incompatible), mod:champions, file/aoa3-epicfight.zip, mod:inventorio (incompatible)
	World Generation: Experimental
	Is Modded: Definitely; Server brand changed to 'forge'
	Type: Dedicated Server (map_server.txt)
	ModLauncher: 9.1.3+9.1.3+main.9b69c82a
	ModLauncher launch target: forgeserver
	ModLauncher naming: srg
	ModLauncher services: 
		 mixin PLUGINSERVICE 
		 eventbus PLUGINSERVICE 
		 slf4jfixer PLUGINSERVICE 
		 object_holder_definalize PLUGINSERVICE 
		 runtime_enum_extender PLUGINSERVICE 
		 capability_token_subclass PLUGINSERVICE 
		 accesstransformer PLUGINSERVICE 
		 runtimedistcleaner PLUGINSERVICE 
		 mixin TRANSFORMATIONSERVICE 
		 fml TRANSFORMATIONSERVICE 
	FML Language Providers: 
		minecraft@1.0
		kotlinforforge@3.2.0
		javafml@null
	Mod List: 
		server-1.18.2-20220404.173914-srg.jar             |Minecraft                     |minecraft                     |1.18.2              |DONE      |Manifest: NOSIGNATURE
		FarmersDelight-1.18.2-1.0.4.jar                   |Farmer's Delight              |farmersdelight                |1.18.2-1.0.4        |DONE      |Manifest: NOSIGNATURE
		grapplemod-1.18.2-1.18.2-v12.3.jar                |grapplemod                    |grapplemod                    |1.18.2-v12.3        |DONE      |Manifest: NOSIGNATURE
		voicechat-forge-1.18.2-2.2.33.jar                 |Simple Voice Chat             |voicechat                     |1.18.2-2.2.33       |DONE      |Manifest: NOSIGNATURE
		EnhancedArmaments-forge-1.18.2-1.1.0.jar          |Enhancedarmaments             |enhancedarmaments             |1.18.2-1.1.0        |DONE      |Manifest: NOSIGNATURE
		sedna-1.18.2-forge-1.0.13+7355dbe.jar             |Sedna for Minecraft           |sedna                         |1.0.13              |DONE      |Manifest: NOSIGNATURE
		AoA3-1.18.2-3.6-Public+Alpha+3.jar                |Advent of Ascension           |aoa3                          |1.18.2-3.6-Public Al|DONE      |Manifest: NOSIGNATURE
		Public+GUI+Announcement-forge-1.18.2-3.2.0.1.jar  |Public GUI Announcement       |publicguiannouncement         |3.2.0.1             |DONE      |Manifest: NOSIGNATURE
		jei-1.18.2-9.7.0.192.jar                          |Just Enough Items             |jei                           |9.7.0.192           |DONE      |Manifest: NOSIGNATURE
		appliedenergistics2-11.0.0-alpha.4.jar            |Applied Energistics 2         |ae2                           |11.0.0-alpha.4      |DONE      |Manifest: NOSIGNATURE
		inventorio-1.18-forge-1.7.0.jar                   |Inventorio                    |inventorio                    |1.7.0               |DONE      |Manifest: NOSIGNATURE
		Botania-1.18.2-431.jar                            |Botania                       |botania                       |1.18.2-431          |DONE      |Manifest: NOSIGNATURE
		wthit-forge-4.8.1.jar                             |wthit                         |wthit                         |4.8.1               |DONE      |Manifest: NOSIGNATURE
		flywheel-forge-1.18-0.6.2.jar                     |Flywheel                      |flywheel                      |1.18-0.6.2          |DONE      |Manifest: NOSIGNATURE
		curios-forge-1.18.2-5.0.7.0.jar                   |Curios API                    |curios                        |1.18.2-5.0.7.0      |DONE      |Manifest: NOSIGNATURE
		BowInfinityFix-1.18.x-rv11.jar                    |Bow Infinity Fix              |bowinfinityfix                |1.18.x-rv11         |DONE      |Manifest: NOSIGNATURE
		Patchouli-1.18.2-67.jar                           |Patchouli                     |patchouli                     |1.18.2-67           |DONE      |Manifest: NOSIGNATURE
		create-mc1.18.2_v0.4.1.jar                        |Create                        |create                        |0.4.1               |DONE      |Manifest: NOSIGNATURE
		Xaeros_Minimap_22.5.0_Forge_1.18.2.jar            |Xaero's Minimap               |xaerominimap                  |22.5.0              |DONE      |Manifest: NOSIGNATURE
		waystones-forge-1.18.2-10.0.0.jar                 |Waystones                     |waystones                     |10.0.0              |DONE      |Manifest: NOSIGNATURE
		gravestone-1.18.2-1.0.2.jar                       |Gravestone Mod                |gravestone                    |1.18.2-1.0.2        |DONE      |Manifest: NOSIGNATURE
		EpicFight-18.0.3-beta.jar                         |Epic Fight                    |epicfight                     |18.0.3              |DONE      |Manifest: NOSIGNATURE
		XaerosWorldMap_1.20.6_Forge_1.18.2.jar            |Xaero's World Map             |xaeroworldmap                 |1.20.6              |DONE      |Manifest: NOSIGNATURE
		markdown_manual-1.18.2-forge-1.2.1+59006bb.jar    |Markdown Manual               |markdown_manual               |1.2.1               |DONE      |Manifest: NOSIGNATURE
		citadel-1.11.2-1.18.2.jar                         |Citadel                       |citadel                       |1.11.2              |DONE      |Manifest: NOSIGNATURE
		Cataclysm-0.17.jar                                |Cataclysm Mod                 |cataclysm                     |1.0                 |DONE      |Manifest: NOSIGNATURE
		AutoRegLib-1.7-53.jar                             |AutoRegLib                    |autoreglib                    |1.7-53              |DONE      |Manifest: NOSIGNATURE
		TravelersBackpack-1.18.2-7.1.0.jar                |Traveler's Backpack           |travelersbackpack             |7.1.0               |DONE      |Manifest: NOSIGNATURE
		StorageDrawers-1.18.2-10.2.0.jar                  |Storage Drawers               |storagedrawers                |10.2.0              |DONE      |Manifest: NOSIGNATURE
		oc2-1.18.2-forge-0.1.7+6a003fc.jar                |OpenComputers II              |oc2                           |0.1.7               |DONE      |Manifest: NOSIGNATURE
		DrawersTooltip-1.18.2-forge-5.0.0.jar             |Drawers Tooltip               |drawerstooltip                |5.0.0               |DONE      |Manifest: NOSIGNATURE
		elevatorid-1.18.2-1.8.4.jar                       |Elevator Mod                  |elevatorid                    |1.18.2-1.8.4        |DONE      |Manifest: NOSIGNATURE
		champions-forge-1.18.2-2.1.3.0.jar                |Champions                     |champions                     |1.18.2-2.1.3.0      |DONE      |Manifest: NOSIGNATURE
		guardvillagers-1.18.2.1.4.1.jar                   |Guard Villagers               |guardvillagers                |1.4.1               |DONE      |Manifest: NOSIGNATURE
		balm-3.0.2+0.jar                                  |Balm                          |balm                          |3.0.2+0             |DONE      |Manifest: NOSIGNATURE
		JustEnoughResources-1.18.2-0.14.1.160.jar         |Just Enough Resources         |jeresources                   |0.14.1.160          |DONE      |Manifest: NOSIGNATURE
		BetterShieldsMC1.18.1-1.0.1.jar                   |Better Shields                |bettershields                 |1.0.1               |DONE      |Manifest: NOSIGNATURE
		cloth-config-6.2.57-forge.jar                     |Cloth Config v4 API           |cloth_config                  |6.2.57              |DONE      |Manifest: NOSIGNATURE
		forge-1.18.2-40.0.52-universal.jar                |Forge                         |forge                         |40.0.52             |DONE      |Manifest: 84:ce:76:e8:45:35:e4:0e:63:86:df:47:59:80:0f:67:6c:c1:5f:6e:5f:4d:b3:54:47:1a:9f:7f:ed:5e:f2:90
		goprone-1.18.2-1.7.0.jar                          |GoProne                       |goprone                       |1.18.2-1.7.0        |DONE      |Manifest: NOSIGNATURE
		Quark-3.2-348.jar                                 |Quark                         |quark                         |3.2-348             |DONE      |Manifest: NOSIGNATURE
		veinmining-forge-1.18.2-0.18.jar                  |Vein Mining                   |veinmining                    |1.18.2-0.18         |DONE      |Manifest: NOSIGNATURE
		domesticationinnovation-1.2.0-1.18.2.jar          |Domestication Innovation      |domesticationinnovation       |1.2.0               |DONE      |Manifest: NOSIGNATURE
	Crash Report UUID: 31c53a05-c897-48e6-85e7-2cfe0b26704e
	FML: 40.0
	Forge: net.minecraftforge:40.0.52
```  
Since I'm also a developer, I dived into the code and found [the root cause](https://github.com/Yesssssman/epicfightmod/blob/1.18.2/src/main/java/yesman/epicfight/world/entity/ai/goal/ChasingGoal.java#L140)  
Although I'm not clear about in which situation will `net.minecraft.world.entity.LivingEntity.Mob.getTarget` returns a null value, the field `net.minecraft.world.entity.LivingEntity.Mob.target` is annotated with `Nullable`, which means that the null value is an expected behavior. I've searched the usage of `Mob.getTarget` in the project and in most cases, the nullability of the result is checked right after the call.  
So maybe the fix would be as simple as adding something like below after the call
```java
if (livingentity == null)
    return;
```

**Comment by Yesssssman** [comment]
Both ChasingGoal#canUse and ChasingGoal#canContinueToUse checks if the target is null. I guess some mod stores the entity's target null during the Goal#tick method and it could bring other problems.

**Comment by truemogician** [comment]
Ummm, so there's a concurrency conflict. No wonder the occurrence of the problem is unpredictable. Since the conflict comes from a different thread. thus the crash report can't record the real problem. What a knotty problem!

**Comment by Yesssssman** [comment]
do you have any idea what mod causes this crash? it may have something to do with mob AI.

**Comment by Yesssssman** [comment]
Forget what I said. I found that the vanilla AI also removes the attack target. I'd just simply use your suggestion.

**Comment by truemogician** [comment]
I've just looked through forge's mapped code, two other subclass of `Goal` also face the same situation: `net.minecraft.world.entity.ai.goal.MeleeAttackGoal` and `net.minecraft.world.entity.ai.goal.RangedBowAttackGoal`. They both checked the nullability of `target` right after the call even though it's already checked in `canUse` and `canContinueToUse`. So maybe it's just normal situation that the `target` gets removed between calling `canUse` and `tick`.

**Comment by truemogician** [comment]
Besides, I followed the call stack in the crash report bottom-up, neither `Goal.canUse`  nor `Goal.canContinueToUse` is called before the exception point. That is to say, `Goal.tick` doesn't depend on `Goal.canUse` and `Goal.canContinueToUse`. Maybe it used to be so in some early versions, but here in 1.18.2, they do seem to be rather irrelevant.

**Comment by Yesssssman** [comment]
Fixed in 18.1.5
